File: src/main/java/com/example/testreactive/service/PersonService.java
package com.example.testreactive.service;

import com.example.testreactive.dto.PersonDto;
import reactor.core.publisher.Flux;

public interface PersonService {
    Flux<PersonDto> getRandomPersons();
    Flux<PersonDto> getFastRandomPersons();
    Flux<PersonDto> getSlowRandomPersons();

    Flux<PersonDto> getStartsFistName(String firstNameLetter);
}

File: src/main/kotlin/com/example/testreactive/service/PersonService.kt
package com.example.testreactive.service

import com.example.testreactive.dto.PersonDto

interface PersonService {
    suspend fun getRandomPersons(): List<PersonDto>
    suspend fun getFastRandomPersons(): List<PersonDto>
    suspend fun getSlowRandomPersons(): List<PersonDto>
    suspend fun getStartsFistName(firstNameLetter: String): List<PersonDto>
}
